_source/_posts/2022-02-25-thymeleaf-security.md

```diff
@@ -0,0 +1,670 @@
+    </div>
+    <form action="#" th:action="@{/quiz}" th:object="${quiz}"
+          method="post" class="col-md-4 fw-light">
+        <ul>
+            <li th:errors="*{answer}" />
+        </ul>
+        <div class="col-md-12">
+            <h3>What is Thymeleaf?</h3>
+        </div>
+        <div class="col-md-12 form-check">
+            <input class="form-check-input" type="radio" th:field="*{answer}" value="A" id="check-1-1"/>
+            <label class="form-check-label" for="check-1-1">
+                <strong>A.</strong> A server-side Java template engine
+            </label>
+        </div>
+        <div class="col-md-12 form-check">
+            <input class="form-check-input" type="radio" th:field="*{answer}" value="B" id="check-1-2"/>
+            <label class="form-check-label" for="check-1-2">
+                <strong>B.</strong> A markup language
+            </label>
+        </div>
+        <div class="col-md-12 form-check">
+            <input class="form-check-input" type="radio" th:field="*{answer}" value="B" id="check-1-3"/>
+            <label class="form-check-label" for="check-1-3">
+                <strong>C.</strong> A web framework
+            </label>
+        </div>
+        <div class="col-md-12 mt-4 mb-4">
+            <p>Your CSRF token is:<span th:text="${_csrf.token}"/></p>
+        </div>
+        <div class="col-md-12">
+            <button type="submit" class="btn btn-primary">Submit</button>
+        </div>
+    </form>
+</div>
+</body>
+<!--/*/ <th:block th:include="footer :: footer"/> /*/-->
+</html>
+```
+
+The CSRF token is available as a request attribute and will be displayed in the `quiz.html` template for learning purposes.
+
+
+Add a template for the quiz result with the name `result.html`:
+
+```html
+<html xmlns:th="http://www.thymeleaf.org">
+<head>
+    <title>Thymeleaf Quiz Submission</title>
+    <!--/*/ <th:block th:include="head :: head"/> /*/-->
+</head>
+<body id="samples">
+<div th:replace="menu :: menu"></div>
+<div class="container" id="content">
+    <div class="text-center">
+        <i class="bi-balloon-heart-fill" style="font-size: 6rem; color: green;" th:if=${quiz.answer=='A'}></i>
+        <i class="bi-x-circle-fill" style="font-size: 6rem; color: red;" th:unless=${quiz.answer=='A'}></i>
+        <div class="panel mt-4 text-center">
+            <div class="panel-body">
+                <h4>Your selected answer is <strong>
+                    <span th:text="${quiz.answer}"></span>
+                </strong></h4>
+                <p th:if=${quiz.answer=='A'}>Good Job!</p>
+            </div>
+        </div>
+        <div class="panel mt-4 text-center" th:unless=${quiz.answer=='A'}>
+            <div class="panel-body">
+                <h4>The right answer is <strong>A:</strong></h4>
+                <h4>A server-side Java template engine</h4>
+            </div>
+        </div>
+    </div>
+</div>
+</body>
+<!--/*/ <th:block th:include="footer :: footer"/> /*/-->
+</html>
+```
```

Comment:
It'd be nice to have a link back to the quiz in this template? It could say something like `Try again?`

Reply:
Added the link